### PR TITLE
feat (packages/codemod): Add more automated transformations for 4.0.

### PR DESCRIPTION
**Pull request by shaper**
Broad brush strokes with generated test cases. There is likely more to do to transform with precision e.g. making sure affected sites are in the expected ctor, method, etc. (see `baseURL` for example).

### CODE REVIEW

File: packages/codemod/package.json [hunk 1]
```diff
@@ -8,6 +8,7 @@
         "test/dist/**/*",
         "CHANGELOG.md"
     ],
+    "type": "module",
```
Comment (lgrammel) [inline packages/codemod/package.json:11]: why is this needed? will there be any downstream complications for users?

**Review by lgrammel — APPROVED** [review]

File: packages/codemod/package.json (hunk 1 above)
Comment (shaper) [reply packages/codemod/package.json:11]: https://vite.dev/guide/troubleshooting.html#vite-cjs-node-api-deprecated has more background. I think it's basically low priority to make the change in any of our dev-facing libraries intended for import, and could cause issues for some folks. For this particular module given intended use it's probably ok, though admittedly deviant standard-wise.

Comment (shaper) [reply packages/codemod/package.json]: Moved back to cjs for consistency, triggered by my finding that ESM required further edits for local iteration running `codemod` binary (now operational, which is nice). We can revisit later.
